api/workflow/v1alpha1/types.go

```diff
@@ -295,6 +313,40 @@ type Script struct {
 	Source  string   `json:"source"`
 }
 
+// ResourceTemplate is a template subtype to manipulate kubernetes resources
+type ResourceTemplate struct {
+	// Action is the action to perform to the resource.
+	// Must be one of: create, apply, delete
+	Action string `json:"action"`
+
+	// Manifest contains the kubernetes manifest
+	Manifest string `json:"manifest"`
+
+	// WaitCondition is a label selector expression which describes the conditions
+	// of the k8s resource in which it is acceptable to proceed to the following step
+	WaitCondition string `json:"waitCondition,omitempty"`
```

Comment:
Can we please rename this because this is confusing that this is the success condition

Reply:
Ok. the agreement seems to be: `successCondition`